testdata/p4_14_samples/switch_20160226/includes/intrinsic.p4

```diff
@@ -19,44 +19,46 @@ header_type ingress_intrinsic_metadata_t {
         resubmit_flag : 1;              // flag distinguishing original packets
                                         // from resubmitted packets.
 
-        ingress_global_tstamp : 48;     // global timestamp (ns) taken upon
+        ingress_global_timestamp : 48;  // global timestamp (ns) taken upon
                                         // arrival at ingress.
 
         mcast_grp : 16;                 // multicast group id (key for the
                                         // mcast replication table)
-
+#ifdef INCLUDE_OLD_INTRINSIC_METADATA_FIELDS
         deflection_flag : 1;            // flag indicating whether a packet is
                                         // deflected due to deflect_on_drop.
         deflect_on_drop : 1;            // flag indicating whether a packet can
                                         // be deflected by TM on congestion drop
-
-        enq_qdepth : 19;                // queue depth at the packet enqueue
-                                        // time.
-        enq_tstamp : 32;                // time snapshot taken when the packet
-                                        // is enqueued (in nsec).
-        enq_congest_stat : 2;           // queue congestion status at the packet
-                                        // enqueue time.
-
-        deq_qdepth : 19;                // queue depth at the packet dequeue
-                                        // time.
-        deq_congest_stat : 2;           // queue congestion status at the packet
-                                        // dequeue time.
-        deq_timedelta : 32;             // time delta between the packet's
-                                        // enqueue and dequeue time.
-
         mcast_hash : 13;                // multicast hashing
+#endif // INCLUDE_OLD_INTRINSIC_METADATA_FIELDS
         egress_rid : 16;                // Replication ID for multicast
         lf_field_list : 32;             // Learn filter field list
         priority : 3;                   // set packet priority
     }
 }
 metadata ingress_intrinsic_metadata_t intrinsic_metadata;
 
-#define _ingress_global_tstamp_     intrinsic_metadata.ingress_global_tstamp
+#define _ingress_global_tstamp_     intrinsic_metadata.ingress_global_timestamp
 
 action deflect_on_drop() {
+#ifdef INCLUDE_OLD_INTRINSIC_METADATA_FIELDS
     modify_field(intrinsic_metadata.deflect_on_drop, 1);
+#endif // INCLUDE_OLD_INTRINSIC_METADATA_FIELDS
+}
+
+header_type queueing_metadata_t {
+    fields {
+        enq_timestamp : 48;             // time snapshot taken when the packet
+                                        // is enqueued (in microsec).
+        enq_qdepth : 16;                // queue depth at the packet enqueue
```

Comment:
Well, if they are here or in intrinsic_metadata it's rather the same thing; they won't be handled correctly by the compiler if they are expected to have values when the pipeline starts. We still need to move them to stdandard_metadata in v1model.

Reply:
I agree, but at least this is the name of the metadata "header" that simple_switch documentation has said that these fields must be defined in a P4_14 program for several years now, here: https://github.com/p4lang/behavioral-model/blob/master/docs/simple_switch.md

Reply:
Also, these fields _are_ initialized by simple_switch today.

Reply:
Yes, but the way the P4-14 to P4-16 translates this struct will not enable it to use the simple_switch values. We essentially need to put these fields in standard_metadata and make the translator refer to them.

Reply:
The simple_switch documentation about queueing_metadata is the reason I asked about it in this comment earlier: https://github.com/p4lang/p4c/pull/1704#issuecomment-459170332

You replied you had no example programs that used queueing_metadata: https://github.com/p4lang/p4c/pull/1704#issuecomment-459175619

I just did some grep'ing through the code and found this test program that uses queueing_metadata, that was committed to the code in 2016 and hasn't been modified since then: https://github.com/p4lang/p4c/blob/master/testdata/p4_14_samples/queueing.p4

Please let me know if there is something you would like changed in this PR, though.  I am fine with whatever on this P4_14 intrinsic_metadata and queueing_metadata stuff.  It has been documented for years as working this way, then broken some time around Nov 2018.  Just looking for a path to getting back to working here.

Reply:
By the way, if part of the path to getting things working is to change documentation that has been the same for a couple of years, I have no problem with that, either.  Just nice if the documentation and implementation are consistent with each other, is all.